src/ngResource/resource.js

```diff
@@ -431,16 +431,20 @@ angular.module('ngResource', ['ng']).
             }
             if (!(new RegExp("^\\d+$").test(param)) && param &&
               (new RegExp("(^|[^\\\\]):" + param + "(\\W|$)").test(url))) {
-              urlParams[param] = true;
+              urlParams[param] = { isQueryParam: (new RegExp("\\?(.*)\\:" + param)).test(url) };
```

Comment:
No need to make it too strict either :)

IMO, `new RegExp('\\?(.*):' + param + '(\\W|$)')` (or `new RegExp('\\?(?:.*):' + param + '(?:\\W|$)')`) is fine.